ABDataCollectionCore.js

```diff
@@ -2605,6 +2614,38 @@ module.exports = class ABDataCollectionCore extends ABMLClass {
       return result;
    }
 
+   updateRelationalDataFromLinkDC(objectId, rowData) {
+      const dcLink = this.datacollectionLink;
+      const cursorLink = dcLink?.getCursor();
+
+      // Add the new data that just relate to the Link DC
+      if (
+         dcLink?.datasource.id == objectId &&
+         cursorLink &&
+         cursorLink.id == rowData?.id
+      ) {
+         const obj = this.datasource;
+         const linkedField = this.fieldLink;
+         let relatedData = rowData[linkedField.fieldLink.relationName()];
+         if (relatedData && !Array.isArray(relatedData))
+            relatedData = [relatedData];
+
+         (relatedData ?? []).forEach((item) => {
+            if (item == null) return;
+
+            if (!this.__dataCollection.exists(item[obj.PK()])) {
+               // QUESTION: Should we .find to get fully info here ?
+               const newItem = this.AB.cloneDeep(item);
```

Comment:
To answer your question: Yes, you need to perform a .find to get the fully populated data for use in your DataCollection.

You could look at reusing the `this.datasource.model().staleRefresh()` as that does some buffering to only make 1 call even if you are asking for 10 new items.